### PR TITLE
Address Auth Edge Case

### PR DESCRIPTION
**Pull request by pablonyx**
## How Has This Been Tested?

[Describe the tests you ran to verify your changes]

## Backporting (check the box to trigger backport action)

Note: You have to check that the action passes, otherwise resolve the conflicts manually and tag the patches.

- [ ] This PR should be backported (make sure to check that the backport attempt succeeds)
- [ ] [Optional] Override Linear Check

### CODE REVIEW

File: backend/onyx/auth/users.py [hunk 1]
```diff
@@ -420,7 +420,7 @@ async def oauth_callback(
             except exceptions.UserNotExists:
                 try:
                     # Attempt to get user by email
-                    user = await self.get_by_email(account_email)
+                    user = cast(User, await self.user_db.get_by_email(account_email))
```
Comment (evan-onyx) [inline backend/onyx/auth/users.py:423]: Slightly prefer asserting non-None rather than casting here

**Review by evan-danswer — APPROVED** [review]
LGTM with small suggestion
